07-routing-gossip.md

```diff
@@ -1136,6 +1157,200 @@ A->D's `update_add_htlc` message would be:
 And D->C's `update_add_htlc` would again be the same as B->C's direct payment
 above.
 
+## Liquidity Ads
+
+Liquidity ads allow nodes to announce their willingness to provide funding to
+other nodes for a fee. Sellers advertise various rates for the liquidity they
+are selling depending on the guarantees offered to buyers.
+
+1. subtype: `will_fund_lease_rates`
+2. data:
+    * [`will_fund_lease_rates_tlvs`:`lease_rates`]
+
+1. `tlv_stream`: `will_fund_lease_rates_tlvs`
+2. types:
+    1. type: 1 (`basic_funding_leases`)
+    2. data:
+        * [`...*basic_funding_lease`:`funding_lease_rates`]
+    1. type: 3 (`duration_based_funding_leases`)
+    2. data:
+        * [`...*duration_based_funding_lease`:`funding_lease_rates`]
+
+1. subtype: `request_funds`
+2. data:
+    * [`u64`:`requested_sats`]
+    * [`byte`:`lease_type`]
+    * [`funding_lease`:`funding_lease`]
+
+1. `lease_type`: 1 (`basic_funding_lease`)
+2. data:
+    * [`u32`:`min_lease_amount_sat`]
+    * [`u32`:`max_lease_amount_sat`]
+    * [`funding_lease_fee`:`funding_lease_fee`]
+
+1. `lease_type`: 3 (`duration_based_funding_lease`)
```

Comment:
I included this second lease type to show how supporting multiple lease types is done, but I'm not convinced it should be included in the first version. Should we leave it out for now?

Reply:
Done in https://github.com/lightning/bolts/pull/1153/commits/d30aa2b2ab0980e3a58a58bd3b8341c028a67bd3: readers should have a look at the first commit to see how multiple `lease_types` can be defined.